### PR TITLE
Fix for docker-compose

### PR DESCRIPTION
**Pull request by Victoremepunto**
- Add a mandatory environment variable for PostgreSQL container to
 run
 - Remove unnecessary network section
 - Add a note on README to include the upload-service pre-requisite

### CODE REVIEW

**Comment by Victoremepunto** [comment]
Not sure about the pre-requisite section, specially since `upload-service` is archived, though there's no mention which service replaces it (I think it's ingress go?)

File: README.md [hunk 1]
```diff
@@ -88,6 +88,10 @@ SPANDX_CONFIG=$(pwd)/compliance-backend.js ../insights-proxy/scripts/run.sh
 
 ### Option 3: Docker Development setup
 
+## pre-requisites:
+
+There should be an [upload-service](https://github.com/RedHatInsights/insights-upload) container available in the same network the `rails` container runs.
```
Comment (akofink) [inline README.md:93]: It only really requires access to the same kafka instance that an upload instance is producing on.

File: docker-compose.yml [hunk 2]
```diff
@@ -95,7 +97,3 @@ services:
   redis:
     image: redis:latest
 
-networks:
-  default:
-    external:
-      name: docker_default # should match insights-upload's network name
```
Comment (akofink) [inline docker-compose.yml:101]: Without this defined as an external network, we won't be able to access kafka, defined in its own docker-compose, by its service name (i.e. `kafka`). https://github.com/RedHatInsights/insights-ingress-go/blob/master/docker-compose.yml

File: docker-compose.yml [hunk 3]
```diff
@@ -3,6 +3,8 @@ services:
   db:
     image: postgres
     restart: on-failure
+    environment:
+        - POSTGRES_PASSWORD=postgres
```
Comment (akofink) [inline docker-compose.yml:7]: I'm not sure why it's worked this whole time without this. The docs do say it's mandatory. Does it not work for you?

Comment (bastilian) [reply docker-compose.yml:7]: This seems to have only changed recently. If the DB was initialised before that it probably is not needed, but bootstrapping fresh, did also require the password for me. https://github.com/docker-library/postgres/commit/f1bc8782e7e57cc403d0b32c0e24599535859f76

**Comment by akofink** [comment]
@Victoremepunto since you're working on podman now, shall we close this one?

**Comment by Victoremepunto** [comment]
> @Victoremepunto since you're working on podman now, shall we close this one?

I'm not sure we don't need fixes for docker-compose, since the podman scripts won't necessarily replace docker-compose immediately, I was discussing with @bastilian about this subject, and I think it is in our best interest to keep maintaining docker-compose for a while, even if we get to a point where we get a functional podman based alternative.

**Comment by akofink** [comment]
Makes sense. There are some unaddressed comments inline.

File: docker-compose.yml (hunk 3 above)
Comment (akofink) [reply docker-compose.yml:7]: Don't we also need to add this env variable as `'POSTGRESQL_ADMIN_PASSWORD'` to the services which require the DB, like `rails`, `sidekiq`, etc.?

**Comment by Victoremepunto** [comment]
Closed by #527

**Pull request closed without merging**